website/src/_includes/section-menu/documentation.html

```diff
@@ -121,6 +121,14 @@
         <li><a href="{{ site.baseurl }}/documentation/programming-guide/#using-metrics">Using metrics in pipeline</a></li>
       </ul>
     </li>
+    <li class="section-nav-item--collapsible">
+      <span class="section-nav-list-title">Lifecycles</span>
+
+      <ul class="section-nav-list">
+        <li><a href="{{ site.baseurl }}/documentation/programming-guide/#dofn">DoFn lifecycle</a></li>
+        <li><a href="{{ site.baseurl }}/documentation/programming-guide/#source">Source lifecycle</a></li>
```

Comment:
I think this would be good under Pipeline I/O > Developing New I/O Connectors

Reply:
Agree, I've put it in the Source part